### PR TITLE
SNOW-692945: multiple statements support

### PR DESCRIPTION
**Pull request by SimbaGithub**
teamwork issue 1036

This PR is based on refactoring in https://github.com/snowflakedb/libsnowflakeclient/pull/724. Will rebase when that one is merged.
Draft for now as temporarily change included to set the identity sending in login request to ODBC 3.0.1 (changes in client.h and version.h) to run test and will remove the change before merging.
Will need server change to enable multiple statements support on server side for libsfc and PHP when the feature is released.

### CODE REVIEW

File: include/snowflake/client.h [hunk 1]
```diff
@@ -18,7 +18,7 @@ extern "C" {
 /**
  * API Name
  */
-#define SF_API_NAME "C API"
+#define SF_API_NAME "ODBC"
```
Comment (sfc-gh-dstempniak) [inline include/snowflake/client.h:21]: Is it intentional?

File: include/snowflake/version.h [hunk 2]
```diff
@@ -5,6 +5,6 @@
 #ifndef SNOWFLAKE_CLIENT_VERSION_H
 #define SNOWFLAKE_CLIENT_VERSION_H
 
-#define SF_API_VERSION "1.0.13"
+#define SF_API_VERSION "3.0.1"
```
Comment (sfc-gh-dstempniak) [inline include/snowflake/version.h:8]: Is it intentional?

**Review by sfc-gh-dstempniak — APPROVED** [review]

File: include/snowflake/version.h (hunk 2 above)
Comment (sfc-gh-ext-simba-hx) [reply include/snowflake/version.h:8]: same as the conversation above.

File: include/snowflake/client.h (hunk 1 above)
Comment (sfc-gh-dstempniak) [reply include/snowflake/client.h:21]: Thx, didn't notice.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Pull request merged**